### PR TITLE
Add keep-alive ping

### LINKED ISSUE #3113: keep-alive ping on server websocket
To keep reverse proxies like Heroku's from deciding to close the websocket, we should send a message every so often (the interval should be configurable, but Heroku closes the socket after 60 seconds inactive, so something like 45 seconds could be a good default). If we discover that other popular kinds of hosting close after a shorter interval we could make the default shorter.
The message would be an empty ping of some kind.

### PR DESCRIPTION
**Pull request by havocp**
This fixes #3113

### CODE REVIEW

**Comment by havocp** [comment]
It's possibly better to do a separate periodic callback per session so we don't "pause" for a while to send all these pings. But it's also possibly worse since it would add a bunch of periodic callbacks to the io loop, and maybe it's very fast to send all the pings anyway (it probably is...)

So I guess doing it the easy way like this makes sense until we profile.

File: bokeh/server/tornado.py [hunk 1]
```diff
@@ -35,14 +35,26 @@ class BokehTornado(TornadoApplication):
...
+                 io_loop=None,
+                 extra_patterns=None,
+                 keep_alive_milliseconds=None):
         if io_loop is None:
             io_loop = IOLoop.current()
         self._loop = io_loop
 
+        if keep_alive_milliseconds is None:
+            # the default is <60 seconds because that seems like a common timeout
+            keep_alive_milliseconds = 37*1000
```
Comment (bryevdv) [inline bokeh/server/tornado.py]: Why not just make this value a default for the argument?

**Comment by havocp** [comment]
nginx and heroku both default to 60s. The nginx config parameter is proxy_read_timeout (http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_read_timeout ), we could mention in our docs the option to increase that if people want to do less pinging.

**Comment by bryevdv** [comment]
LGTM will merge when passes.

**Comment by bryevdv** [comment]
Not sure why tests are failing.

**Comment by havocp** [comment]
test failure was because keep alive None was no longer valid

**Pull request merged**